homeassistant/components/wled/__init__.py

```diff
@@ -115,26 +84,68 @@
     return True
 
 
+def wled_exception_handler(func):
+    """Decorate WLED calls to handle WLED exceptions.
+
+    A decorator that wraps the passed in function, catches WLED errors,
+    and handles the availability of the device in the data coordinator.
+    """
+
+    async def handler(self, *args, **kwargs):
+        try:
+            return await func(self, *args, **kwargs)
+        except WLEDError as error:
+            _LOGGER.error("Error communicating with API: %s", error)
+            self.coordinator.last_update_success = False
+
+    return handler
+
+
+class WLEDDataUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching WLED data from single endpoint."""
+
+    def __init__(
+        self, hass: HomeAssistant, *, host: str,
+    ):
+        """Initialize global WLED data updater."""
+        self.wled = WLED(host, session=async_get_clientsession(hass))
+
+        async def async_update_data():
```

Comment:
Optional: make this a class method. 

Even more optional: make it a class method on DataUpdateCoordinator that falls back to checking the passed in method.

Reply:
Oh right, that would make implementing these cases even easier 👍

Reply:
done.